File: 2021/10/p1.py
# https://adventofcode.com/2021/day/10

CHAR_MAPPING = { ')': '(', ']': '[', '}': '{', '>': '<' }

POINTS_MAPPING = { ')': 3, ']': 57, '}': 1197, '>': 25137 }

def solve(data):
	points = []

	for entry in data:
		stack = []

		for ch in entry:
			if ch in [ '(', '[', '{', '<' ]:
				stack.append(ch)
			else:
				if stack[-1] != CHAR_MAPPING[ch]:
					points.append(POINTS_MAPPING[ch])
					break
				else:
					stack.pop()

	return sum(points)

def parse(content):
	return [ line for line in content.split('\n') ]

with open('input.txt') as file:
	data = parse(file.read()[:-1])

	print(solve(data))

File: 2015/01/p1.rs
// https://adventofcode.com/2015/day/1

fn main() {
    let raw_data = include_str!("input.txt");

    let result = raw_data
        .chars()
        .map(|x| (x == '(') as i32 * 2 - 1)
        .sum::<i32>();

    println!("{}", result);
}
